Makefile

```
@@ -1,92 +1,30 @@
-
-GROUP ?= all
-FOR_GROUP_TARGET=for-$(GROUP)-target
-
-.DEFAULT_GOAL := all
-
-.PHONY: all
-all: misspell
-
-# Append root module to all modules
-GOMODULES = $(ALL_MODULES)
-
-# Define a delegation target for each module
-.PHONY: $(GOMODULES)
-$(GOMODULES):
-	@echo "Running target '$(TARGET)' in module '$@'"
-	$(MAKE) -C $@ $(TARGET)
-
-# Triggers each module's delegation target
-.PHONY: for-all-target
-for-all-target: $(GOMODULES)
-
-.PHONY: gomoddownload
-gomoddownload:
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="moddownload"
-
-.PHONY: gotest
-gotest:
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="test"
-
-.PHONY: golint
-golint:
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="lint"
-
-.PHONY: golicense
-golicense:
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="license-check"
-
-.PHONY: gofmt
-gofmt:
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="fmt"
-
-.PHONY: gotidy
-gotidy:
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="tidy"
-
-.PHONY: gogenerate
-gogenerate:
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="generate"
-	@$(MAKE) $(FOR_GROUP_TARGET) TARGET="fmt"
-
-.PHONY: gogovulncheck
-gogovulncheck:
-	$(MAKE) $(FOR_GROUP_TARGET) TARGET="govulncheck"
-
-.PHONY: goporto
-goporto:
-	$(MAKE) $(FOR_GROUP_TARGET) TARGET="porto"
-
-# Build a collector based on the Elastic components (generate Elastic collector)
-.PHONY: genelasticcol
-genelasticcol: $(BUILDER)
-	$(BUILDER) --config ./distributions/elastic-components/manifest.yaml
-
-# Validate that the Elastic components collector can run with the example configuration.
-.PHONY: elasticcol-validate
-elasticcol-validate: genelasticcol
-	./_build/elastic-collector-components validate --config ./distributions/elastic-components/config.yaml
-
-.PHONY: builddocker
-builddocker:
-	@if [ -z "$(TAG)" ]; then \
-		echo "TAG is not set. Please provide a tag using 'make builddocker TAG=<tag>'"; \
-		exit 1; \
-	fi
-	@if [ ! -f "_build/elastic-collector-components" ]; then \
-		GOOS=linux $(MAKE) genelasticcol; \
-	fi
-	@if [ -n "$(USERNAME)" ]; then \
-		IMAGE_NAME=$(USERNAME)/elastic-collector-components:$(TAG); \
-	else \
-		IMAGE_NAME=elastic-collector-components:$(TAG); \
-	fi; \
-	docker build -t $$IMAGE_NAME -f distributions/elastic-components/Dockerfile .
+# This currently only works for OS X ARM64.
+.PHONY: install-ocb
+install-ocb:
+	curl --proto '=https' --tlsv1.2 -fL -o ocb \
+	https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/cmd%2Fbuilder%2Fv0.127.0/ocb_0.127.0_darwin_arm64
+	chmod +x ocb
+
+.PHONY: build
+build:
+	./ocb --config builder-config.yaml
+
+.PHONY: tidy
+tidy:
+	cd otelcol-dev && go mod tidy
+
+# This install the otel collector with the name metricsgenreceiver in the GOPATH/bin directory.
+.PHONY: install
+install: build
+	cp ./otelcol-dev/otelcol $(GOPATH)/bin/metricsgenreceiver
+
+# Run the collector against the local ./otelcol.dev.yaml configuration file
+.PHONY: run
+run: install
```

```suggestion
run: build
```

we dont need to install, since we run it from the build dir

---

You don't need to install it. I like to have it because then I can use it anywhere. I do that often with other go apps and there I can just do `go install .` but here that doesn't work.